tests/CppUTestExt/MockExpectedCallTest.cpp

```diff
@@ -413,6 +476,33 @@ TEST(MockExpectedCall, hasNoOutputParameter)
     CHECK_FALSE(call->hasOutputParameter(foo));
 }
 
+TEST(MockExpectedCall, hasOutputParameterOfType)
+{
+    TypeForTestingExpectedFunctionCall object(676789);
+    call->withOutputParameterOfTypeReturning("TypeForTestingExpectedFunctionCall", "foo", &object);
+    MockNamedValue foo("foo");
+    foo.setObjectPointer("TypeForTestingExpectedFunctionCall", &object);
+    CHECK(call->hasOutputParameter(foo));
+}
+
+TEST(MockExpectedCall, hasNoOutputParameterOfTypeSameTypeButInput)
+{
+    TypeForTestingExpectedFunctionCall object(543);
+    call->withParameterOfType("TypeForTestingExpectedFunctionCall", "foo", &object);
+    MockNamedValue foo("foo");
+    foo.setObjectPointer("TypeForTestingExpectedFunctionCall", &object);
+    CHECK_FALSE(call->hasOutputParameter(foo));
+}
+
+TEST(MockExpectedCall, hasNoOutputParameterOfTypeDifferentType)
+{
+    TypeForTestingExpectedFunctionCall object(543);
+    call->withOutputParameterOfTypeReturning("TypeForTestingExpectedFunctionCall", "foo", &object);
+    MockNamedValue foo("foo");
+    foo.setObjectPointer("OtherTypeForTestingExpectedFunctionCall", &object);
```

Comment:
How do you suppose this can happen? I mean, in normal execution, the object pointer is based on the information in withOutputPointerReturning(), so how can the type be different? Can you provoke this testing through the MockSuppor interface?